### DISCUSSION TITLE
Are there any guidelines on which version custom plugins should target?

### DISCUSSION THREAD

**Discussion by nils-a** [quartznet/quartznet#1163] [Q&A] [answered]
I have created a plugin to use my favorite DI container in Quartz.NET and I was wondering whether there are guidelines on compatibility for plugins to Quartz.NET.

I.e. should I target the latest (currently 3.3.2) or should I better target 3.0.0 for maximum compatibility?

**Comment by lahma** [comment, marked as answer]
What API are you targeting specifically?

**Reply by nils-a** [reply]
I'm using `IScheduler.JobFactory` to provide my own implementation of a `IJobFactory`. Nothing fancy in there. (https://github.com/nils-org/Quartz.LightCore/blob/develop/src/Quartz.LightCore/LightCoreResolverJobFactory.cs)

**Reply by lahma** [reply]
That's quite stable API . I think you should target lowest major version you want to support, in this case 3.0.0. There won't be changes to that API until maybe 4.0, if then at all.

**Reply by nils-a** [reply]
Thanks!
